### PR TITLE
Housekeeping (fix small issues, take over stale contributor PRs)

### PR DESCRIPTION
**Pull request by guggero**
This PR is a collection of commits that either fix small things or take over stale contributor PRs (original commit authors are always credited).

Improves:
 - Upgrade linter to latest version that is compatible with `go 1.20.x` and fix new issues found.

Fixes https://github.com/lightningnetwork/lnd/issues/2986.
Fixes https://github.com/lightningnetwork/lnd/issues/3596.
Fixes https://github.com/lightningnetwork/lnd/issues/5418.
Fixes https://github.com/lightningnetwork/lnd/issues/6118.
Fixes https://github.com/lightningnetwork/lnd/issues/7129.
Fixes https://github.com/lightningnetwork/lnd/issues/7077.
Fixes https://github.com/lightningnetwork/lnd/issues/7482.

Replaces:
 - https://github.com/lightningnetwork/lnd/pull/6387
 - https://github.com/lightningnetwork/lnd/pull/6448
 - https://github.com/lightningnetwork/lnd/pull/6958
 - https://github.com/lightningnetwork/lnd/pull/6968
 - https://github.com/lightningnetwork/lnd/pull/7078
 - https://github.com/lightningnetwork/lnd/pull/7249
 - https://github.com/lightningnetwork/lnd/pull/7429
 - https://github.com/lightningnetwork/lnd/pull/7581

### CODE REVIEW

**Review by ellemouton — APPROVED** [review]
Very nice 🔥  LGMT!

**Review by yyforyongyu — COMMENTED** [review]
Nice cleanup! Needs a rebase!

**Comment by guggero** [comment]
Rebased!

File: cmd/lncli/commands.go [hunk 1]
```diff
@@ -302,6 +302,11 @@ var sendCoinsCommand = cli.Command{
 				"must satisfy",
 			Value: defaultUtxoMinConf,
 		},
+		cli.BoolFlag{
+			Name: "force, f",
```
Comment (Roasbeef) [inline cmd/lncli/commands.go:306]: I wonder if this'll end up breaking any bash scripting automation out there? 

As it's essentially a "breaking" change from the PoV of the CLI...

**Review by Roasbeef — APPROVED** [review]
LGTM 🪁

One comment re if we want to go ahead with that CLI change to force confirmation for `sendcoins`.

File: cmd/lncli/commands.go (hunk 1 above)
Comment (guggero) [reply cmd/lncli/commands.go:307]: Ah, good catch! Didn't think about that. I did a quick search and at least RaspiBlitz uses the command in a script: https://github.com/rootzoll/raspiblitz/blob/ff6fafdfc9df84a39f23f1191fe82ad0d42400d6/home.admin/BBcashoutWallet.sh#L102

So I decided to throw in an additional check: `if !ctx.Bool("force") && term.IsTerminal(int(os.Stdout.Fd())) {`

That way the confirmation is only displayed if you are on a plain terminal and the standard out is not redirected:

```shell

$ lncli sendcoins --sweepall bcrt1q26avnvcjwutstwjhc0p8dgenyvjwvzrdfv3e50
Amount: 0
Destination address: bcrt1q26avnvcjwutstwjhc0p8dgenyvjwvzrdfv3e50
Confirm payment (yes/no):

$ lncli sendcoins --sweepall bcrt1q26avnvcjwutstwjhc0p8dgenyvjwvzrdfv3e50 | jq -r 'txid'
e5e49332504cc1619117cc5f13d296d9335ecd25ab2ec43d5e635347212fd9e5

```

**Review by yyforyongyu — COMMENTED** [review]
Looks like it caught an old linter error,
```
for q.pushToActiveQueue(task) {
}
```
that `for` loop looks very weird btw...

**Comment by guggero** [comment]
> Looks like it caught an old linter error,
> 
> ```
> for q.pushToActiveQueue(task) {
> }
> ```
> 
> that `for` loop looks very weird btw...

The code is correct, but I added a comment to make it more clear that we are actually re-trying in a loop. Seems like this is a new false positive added by the new linter version, so I needed to add a `//nolint` directive.

**Review by yyforyongyu — APPROVED** [review]
LGTM🎉

**Comment by guggero** [comment]
@saubyk we're only waiting for @Roasbeef's review because of the comment here: https://github.com/lightningnetwork/lnd/pull/7689#discussion_r1201405805

But since this is more of a user experience question, could you please take a look instead? Also this PR in general (it solves a lot of small issues).

**Comment by saubyk** [comment]
> @saubyk we're only waiting for @Roasbeef's review because of the comment here: [#7689 (comment)](https://github.com/lightningnetwork/lnd/pull/7689#discussion_r1201405805)
> 
> But since this is more of a user experience question, could you please take a look instead? Also this PR in general (it solves a lot of small issues).

I can think of one more project, which can potentially be impacted by this change:
https://github.com/curly60e/pyblock

Although your mitigation should alleviate the concern.

File: docs/release-notes/release-notes-0.17.0.md [hunk 2]
```diff
@@ -72,6 +72,9 @@ package](https://github.com/lightningnetwork/lnd/pull/7356)
   `grpc.keepalive_time_ms=5100` is recommended on the client side (adding 100ms
   to account for slightly different clock speeds).
 
+* [Fixed a bug where we didn't check for correct networks when submitting
+  onchain transactions](https://github.com/lightningnetwork/lnd/pull/6448).
+
 ## Misc
```
Comment (saubyk) [inline docs/release-notes/release-notes-0.17.0.md:78]: `sendcoins` change seems to be missing from the release notes. Given that it's a breaking change (for shell script apps), we must include the change here.

Comment (guggero) [reply docs/release-notes/release-notes-0.17.0.md:78]: Good catch, added!

**Review by saubyk — APPROVED** [review]
LGTM 🧹

**Pull request merged**